localstack/aws/protocol/parser.py

```diff
@@ -294,11 +294,16 @@ def _convert_request_uri_to_regex(request_uri: str) -> Optional[Pattern[str]]:
         """
         if request_uri is None:
             return None
-        # Replace the variable place holder (f.e. {ConnectionId} in "/fuu/{ConnectionId}/bar/")
-        # with a named regex group.
-        request_uri_regex = re.sub(
-            "{(?P<VariableName>[^/]*)}", "(?P<\\g<VariableName>>[^/]*)", request_uri
-        )
+
+        # Replace the variable placeholder (f.e. {ConnectionId} in "/fuu/{ConnectionId}/bar/")
+        # with a named regex group. Allow slashes, in case the placeholder is at the end of the pattern.
+        def _replace(m):
```

Comment:
Note: The logic here could also be implemented with two subsequent regex `sub`s for the two cases. (might have slightly better performance, actually).

Reply:
The change here is quite elaborate and looks good. But to be honest, it might not be necessary at all.
It builds on the assumption that the path parameter matching should avoid parameters with slashes (as the initial implementation suggested with the `[^/]`). However, it might not be necessary at all (sorry for the inconvenience!).

I think it would be better to just ease the limitation in the initial path parameter regex (by switching from `[^/]*` to `.+`). This comes with the risk of a wrong match if a single service specification defines conflicting request URIs, but after a short analysis of the available specs, this doesn't seem to be the case in any service.